midi.py

```diff
@@ -0,0 +1,525 @@
+import os
+import mido
+import numpy as np
+
+from enum import Enum
+from abc import ABC, abstractmethod
+
+"""
+DOCUMENTATION:
```

Comment:
I would argue this documentation should be above the relevant functions. Otherwise this is not displayed in any way in IDEs plus increases the risk of desyncing docs with the implementation. Also, please annotate functions with types where possible. IMO you dont have to describe every combination of parameters, its enough you explain what parameters control

Reply:
DOCUMENTATION moved to DOCUMENTATION.md